### DISCUSSION TITLE
Bedrock does not work.

### DISCUSSION THREAD

**Discussion by diegov12** [danny-avila/LibreChat#4557] [Troubleshooting] [answered]
I've been trying this for the last few days. Access keys have been configured, policies have been attached to the correct user, and models have been requested. I've tested the models within the bedrock playground, fetching a list of models seems to work using the CLI, and the credentials are stored inside of ~/aws. I've tried reading the log, but nothing really seems to point to the problem, although using a screen reader might be the problem there also.
[error-2024-10-26.log](https://github.com/user-attachments/files/17532099/error-2024-10-26.log)
.

**Comment by danny-avila** [comment]
I don't see any relevant logs or screenshots of AWS Bedrock not working.

![image](https://github.com/user-attachments/assets/08db21f5-f0b6-4272-b97e-21edf4d0a40a)

**Comment by diegov12** [comment]
Here's a screenshot I took, hopefully its properly displayed. What other logs should I attach?

<img width="1440" alt="Screenshot 2024-10-28 at 12 20 24 PM" src="https://github.com/user-attachments/assets/978c03d6-5974-40ac-a526-43c07bf35251">

**Reply by danny-avila** [reply]
The logs that are immediately generated following this. Check the error logs file you sent before but send only the parts that are generated after you send the message

**Reply by diegov12** [reply]
OK. Here's what i've managed to get this time.

[error-2024-11-10.log](https://github.com/user-attachments/files/17692982/error-2024-11-10.log)

**Comment by diegov12** [comment]
Tried my credentials in another app, and they work perfectly. So, I can definitely verify everything is set up correctly on the AWS side.

**Reply by ldavis9000aws** [reply]
Have you updated to the latest version of LibreChat? It took a lot of trial and error to get this working initially. What does your .env file for the AWS Bedrock configuration look like? Which variables are you using?

**Comment by diegov12** [comment]
Yes, my version is up-to-date. The only variable i'm using is BEDROCK_AWS_DEFAULT_REGION=us-east-1.

Log returns the same thing still.

**Reply by ldavis9000aws** [reply]
What about your AWS access key and secret key? Where are you defining those variables?

**Comment by diegov12** [comment, marked as answer]
Originally, I wanted the app to source credentials from ~/.aws as my access keys didn't work when defined as variables. I gave it a shot just now by including them as variables, and now it works perfectly.
